components/global/product_switcher_menu/product_switcher_menu.tsx

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {injectIntl, IntlShape} from 'react-intl';
+
+import Icon from '@mattermost/compass-components/foundations/icon';
+
+import {Permissions} from 'mattermost-redux/constants';
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import * as GlobalActions from 'actions/global_actions';
+import AboutBuildModal from 'components/about_build_modal';
+import SystemPermissionGate from 'components/permissions_gates/system_permission_gate';
+import TeamPermissionGate from 'components/permissions_gates/team_permission_gate';
+import MarketplaceModal from 'components/plugin_marketplace';
+import Menu from 'components/widgets/menu/menu';
+
+import {ModalIdentifiers} from 'utils/constants';
+import {useSafeUrl} from 'utils/url';
+import * as UserAgent from 'utils/user_agent';
+
+type Props = {
+    isMobile: boolean;
+    id: string;
+    teamId: string;
+    teamName: string;
+    siteName: string;
+    currentUser: UserProfile;
+    appDownloadLink: string;
+    enableCommands: boolean;
+    enableIncomingWebhooks: boolean;
+    enableOAuthServiceProvider: boolean;
+    enableOutgoingWebhooks: boolean;
+    canManageSystemBots: boolean;
+    canManageIntegrations: boolean;
+    enablePluginMarketplace: boolean;
+    pluginMenuItems: any;
+    intl: IntlShape;
+    firstAdminVisitMarketplaceStatus: boolean;
+};
+
+class ProductSwitcherMenu extends React.PureComponent<Props> {
```

Comment:
Functional component?

Reply:
The menus across webapp will be replaced soon with the common component so for now we are just copying the old menus as we are just splitting them for the purpose of the header.